src/PayU/PaymentsApi/PaymentsV4/Services/RequestBuilder.php

```diff
@@ -0,0 +1,291 @@
+<?php
+
+
+namespace PayU\PaymentsApi\PaymentsV4\Services;
+
+use PayU\Alu\Product;
+use PayU\Alu\Request;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\Authorization\ApplePayToken;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\Authorization\ApplePayToken\ApplePayTokenHeader;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\AuthorizationData;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\AuthorizationRequest;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\Client\BillingData;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\Authorization\CardDetails;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\ClientData;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\Client\DeliveryData;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\Authorization\FxData;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\Client\Billing\IdentityDocumentData;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\Product\Marketplace;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\Authorization\MerchantTokenData;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\ThreeDSecure\MpiData;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\ProductData;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\AirlineInfoData;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\AirlineInfo\FlightSegments;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\StoredCredentialsData;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\ThreeDSecure;
+use PayU\PaymentsApi\PaymentsV4\Entities\Request\AirlineInfo\TravelAgency;
+use PayU\PaymentsApi\PaymentsV4\Exceptions\RequestBuilderException;
+
+class RequestBuilder
+{
+    /**
+     * @param Request $request
+     * @return false|string
+     * @throws RequestBuilderException
+     */
+    public function buildAuthorizationRequest($request)
+    {
+        $authorizationData = new AuthorizationData($request->getOrder()->getPayMethod());
+
+        $authorizationData->setUsePaymentPage($request->getOrder()->getUsePaymentPage());
+        $authorizationData->setInstallmentsNumber($request->getOrder()->getInstallmentsNumber());
+        $authorizationData->setUseLoyaltyPoints($request->getOrder()->getUseLoyaltyPoints());
+        $authorizationData->setLoyaltyPointsAmount($request->getOrder()->getLoyaltyPointsAmount());
+        $authorizationData->setCampaignType($request->getOrder()->getCampaignType());
+
+        if ($request->getCard() !== null &&
```

Comment:
It seems like this big "if" is just for builing $authorizationData.
it would help with code reading to extract this in a separate private method that would do just this, compute and return $authorizationData.

This could also work for AirlineInfoData but that has smaller logic

Reply:
done